1. Write a function that inputs a number and prints the multiplication table of that number

In [1]:
input_num = input("Enter a number")

for num in range(1,11):
    "num= num*int(input_num)"
    print(input_num,"*",num ,"=",num*float(input_num))

Enter a number45
45 * 1 = 45.0
45 * 2 = 90.0
45 * 3 = 135.0
45 * 4 = 180.0
45 * 5 = 225.0
45 * 6 = 270.0
45 * 7 = 315.0
45 * 8 = 360.0
45 * 9 = 405.0
45 * 10 = 450.0


2. Write a program to print twin primes less than 1000. If two consecutive odd numbers are
   both prime then they are known as twin primes

In [2]:
for i in range(1,100):
    if(i%2 !=0):
        print((i,i+2))

(1, 3)
(3, 5)
(5, 7)
(7, 9)
(9, 11)
(11, 13)
(13, 15)
(15, 17)
(17, 19)
(19, 21)
(21, 23)
(23, 25)
(25, 27)
(27, 29)
(29, 31)
(31, 33)
(33, 35)
(35, 37)
(37, 39)
(39, 41)
(41, 43)
(43, 45)
(45, 47)
(47, 49)
(49, 51)
(51, 53)
(53, 55)
(55, 57)
(57, 59)
(59, 61)
(61, 63)
(63, 65)
(65, 67)
(67, 69)
(69, 71)
(71, 73)
(73, 75)
(75, 77)
(77, 79)
(79, 81)
(81, 83)
(83, 85)
(85, 87)
(87, 89)
(89, 91)
(91, 93)
(93, 95)
(95, 97)
(97, 99)
(99, 101)


3. Write a program to find out the prime factors of a number. Example: prime factors of 56 -
   2, 2, 2, 7

In [4]:
import math

def primefactors(n):


    while(n % 2 == 0):
        print(2)
        n=n/2
    
    for i in range(3,int(math.sqrt(n))+1 , 2):
        while(n % i == 0):
            print(i)
            n=n/i
            
        
    if (n > 2):
        print(n)
        
        
        
        
n=input("Enter a number")
primefactors(int(n))
        
        


Enter a number87
3
29.0


4. Write a program to implement these formulae of permutations and combinations.
   Number of permutations of n objects taken r at a time: p(n, r) = n! / (n-r)!. Number of
   combinations of n objects taken r at a time is: c(n, r) = n! / (r!*(n-r)!) = p(n,r) / r!

In [5]:
def permutation_formul():
  n = input("Enter a number :")
  r = input("Enter second number :")

  print (n)
  print (r)

  n_fact=1
  nminr_fact=1

  for i in range (1,(int(n)-int(r))):
    nminr_fact = nminr_fact * i

  for i in range (1,int(n)+1):
    n_fact = n_fact * i
    

  print ("result :" ,(n_fact/nminr_fact))


permutation_formul()

Enter a number :87
Enter second number :56
87
56
result : 7.946218939437255e+99


5. Write a function that converts a decimal number to binary number

In [6]:
def decimal_to_binary(n):
    
    arr = [0]*n
    
    i=0
    if(n > 0):
        while (n > 0):
            arr[i] = n % 2
            n = int(n/2)
            i+=1
        arr.reverse()
        print(arr)
        
    
n= input("Enter a number to convert to binary :")
decimal_to_binary(int(n))

Enter a number to convert to binary :76
[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0]


6. Write a function cubesum() that accepts an integer and returns the sum of the cubes of
   individual digits of that number. Use this function to make functions PrintArmstrong() and
   isArmstrong() to print Armstrong numbers and to find whether is an Armstrong number.

In [7]:
lst = input("Enter a number :")
num = int(lst)
lst = list(lst)
print(lst)

def cubesum(lst):
    sum = 0
    for i in range(len(lst)):
        lst[i]=int(lst[i])**3
        sum +=int(lst[i])
    print(lst)
    print(sum)

    
    def isArmstrong():
        if(num==sum):
            print("armstrong number")
        else:
            print("Not a armstrong number")

    
    isArmstrong()


    

cubesum(lst)  





Enter a number :56
['5', '6']
[125, 216]
341
Not a armstrong number


7. Write a function prodDigits() that inputs a number and returns the product of digits of that
number.

In [8]:
def prodDigits(num):
    
    num = list(num)
    prod = 1;
    for i in range(len(num)):
        
        prod *= int(num[i])
    print(prod)
        

prod = input("Enter a number")

prodDigits(prod)

Enter a number87
56


8. If all digits of a number n are multiplied by each other repeating with the product, the one
digit number obtained at last is called the multiplicative digital root of n. The number of
times digits need to be multiplied to reach one digit is called the multiplicative
persistance of n.
Example: 86 -> 48 -> 32 -> 6 (MDR 6, MPersistence 3)
341 -> 12->2 (MDR 2, MPersistence 2)
Using the function prodDigits() of previous exercise write functions MDR() and
MPersistence() that input a number and return its multiplicative digital root and
multiplicative persistence respectively

In [2]:
def prodDigits(num):
    
    num = list(num)
    prod = 1;
    
    for i in range(len(num)):
        print(num[i])
        
        prod *= int(num[i])
        
        prod_1 = 1
        
        if prod > 9:
            
            while(prod > 10 ):
                
                prod_1 *= prod_1 * (prod % 10)
                print("inside while prod_1",prod_1)
                prod = prod // 10
                print("inside while prod",prod)
                
                

            
            
            
    print(prod)


def MPersistence(num):
    
    if(len(num)!=1):
        count = 0
        prodDigits(num)
        count = count+1
    
    
    print(count)
    



   
    
        
prod = input("Enter a number :")

MPersistence(prod)

Enter a number :456
4
5
inside while prod_1 0
inside while prod 2
6
inside while prod_1 2
inside while prod 1
1
1


9. Write a function sumPdivisors() that finds the sum of proper divisors of a number. Proper
divisors of a number are those numbers by which the number is divisible, except the
number itself. For example proper divisors of 36 are 1, 2, 3, 4, 6, 9, 18

In [ ]:
def sumPdivisors(num):
    
    lst = []
    for i in range(int(num)):
        
        div = int(num)%(i+1)
        
        if(div == 0 and  (i +1) != int(num)):
            lst.append(i+1)
        
    print(lst)
    
    sum = 0
    for i in lst:
        sum += i
        
    print("Sum of divisors",sum)
div = input("Enter a number")

sumPdivisors(div)

10. A number is called perfect if the sum of proper divisors of that number is equal to the
number. For example 28 is perfect number, since 1+2+4+7+14=28. Write a program to
print all the perfect numbers in a given range

In [ ]:
def perfectNum(num):
    
    lst = []
    sum = 0
    for i in range(int(num)):
        
        div = int(num)%(i+1)
        
        if(div == 0 and  (i +1) != int(num)):
            lst.append(i+1)
        
    for j in lst:
        print(j)
        sum += j
           
    
    if (sum == int(num)):
        print("Perfect Number")
    
    print("Sum ",sum)   
    print(lst)
    
    

num = input("Enter a number:")
perfectNum(num)

11. Two different numbers are called amicable numbers if the sum of the proper divisors of
each is equal to the other number. For example 220 and 284 are amicable numbers.
Sum of proper divisors of 220 = 1+2+4+5+10+11+20+22+44+55+110 = 284
Sum of proper divisors of 284 = 1+2+4+71+142 = 220
Write a function to print pairs of amicable numbers in a range

In [21]:
def amicable_Num(num):
    
    lst = []
    
    sum = 0
    for i in range(int(num)):
        
        div = int(num)%(i+1)
        
        if(div == 0 and  (i +1) != int(num)):
            lst.append(i+1)
        
    for j in lst:
        """print(j)"""
        sum += j
           
    
    
        print("Amicable Numbers: ")
        print(j," ",sum)   
       
        """print(lst)"""
    
    

num = input("Enter a number:")

amicable_Num(num)

Enter a number:678
Amicable Numbers: 
1   1
Amicable Numbers: 
2   3
Amicable Numbers: 
3   6
Amicable Numbers: 
6   12
Amicable Numbers: 
113   125
Amicable Numbers: 
226   351
Amicable Numbers: 
339   690


12. Write a program which can filter odd numbers in a list by using filter function

In [ ]:
def oddFilter(lst):
    
    odd =[]
    for i in lst:
        
        if( i%2 != 0):
            odd.append(i)
            
    print("Odd Number :",odd)        
lst = []

for i in range(1,200):
    lst.append(i)
    
filter(oddFilter(lst), lst)

13. Write a program which can map() to make a list whose elements are cube of elements in
a given list

In [ ]:

def cubList(lst):
    
    cube = []
    
    
    for i in lst:
        i = i ** 3
        cube.append(i)
    
    print("Cube List :", cube)
    
lst= []

for i in range(1,10):
    
    lst.append(i)
    
print(lst)

map(cubList(lst),lst)

Write a program which can map() and filter() to make a list whose elements are cube of
even number in a given list

In [ ]:
lst = []
even = []
cube =[]
for i in range(1, 20):
    lst.append(i)
    
def even_lst():
    
   
    for i in lst:
        if (i % 2 == 0):
            even.append(i)
    
    print ("Even List :",even)
            

def cube_lst():
    
    for i in even:
        i = i ** 3
        cube.append(i)
    print("Cube List :",cube)
        
filter(even_lst(),lst)
map(cube_lst(),even)